src/lamplib/src/genny/tasks/yaml_linter.py

```diff
@@ -70,8 +73,56 @@ def check_required_fields(yaml_path):
         if field not in workload:
             if field == "Keywords" and yaml_path in GRANDFATHERED_WORKLOADS_WITHOUT_KEYWORDS:
                 continue
+            if (
```

Comment:
Suggestion: Would it make sense at this stage to create a `Workload` model class and put all this vaidation into a method on the instance? 

I would also suggest that Pydantic might help (although it depends on whether it is complicated enough at this stage).